src/Microsoft.DotNet.Darc/DarcLib/Helpers/IDependencyFileManager.cs

```diff
@@ -23,33 +23,34 @@ Task AddDependencyAsync(
         DependencyDetail dependency,
         string repoUri,
         string? branch,
+        string? mapping = null,
         bool versionDetailsOnly = false,
         bool? repoHasVersionDetailsProps = null);
 
-    Task RemoveDependencyAsync(string dependencyName, string repoUri, string branch, bool repoIsVmr = false, bool? repoHasVersionDetailsProps = null);
+    Task RemoveDependencyAsync(string dependencyName, string repoUri, string branch, string? mapping = null, bool? repoHasVersionDetailsProps = null);
 
     Dictionary<string, HashSet<string>> FlattenLocationsAndSplitIntoGroups(Dictionary<string, HashSet<string>> assetLocationMap);
 
     List<(string key, string feed)> GetPackageSources(XmlDocument nugetConfig, Func<string, bool>? filter = null);
 
-    Task<VersionDetails> ParseVersionDetailsXmlAsync(string repoUri, string branch, bool includePinned = true);
+    Task<VersionDetails> ParseVersionDetailsXmlAsync(string repoUri, string branch, bool includePinned = true, string? mapping = null);
 
-    Task<JObject> ReadDotNetToolsConfigJsonAsync(string repoUri, string branch, bool repoIsVmr);
+    Task<JObject> ReadDotNetToolsConfigJsonAsync(string repoUri, string branch, string? mapping = null);
 
     /// <summary>
     /// Get the tools.dotnet section of the global.json from a target repo URI
     /// </summary>
     /// <param name="repoUri">repo to get the version from</param>
     /// <param name="commit">commit sha to query</param>
-    Task<SemanticVersion> ReadToolsDotnetVersionAsync(string repoUri, string commit, bool repoIsVmr);
+    Task<SemanticVersion> ReadToolsDotnetVersionAsync(string repoUri, string commit, string? mapping = null);
```

Comment:
I think it might be quite confusing what is "mapping" and what does it do.
I think this class should not know about "mappings" rather it would make more sense if we had something like `relativePath` so that it would be obvious we want to work with version files under in a specific directory.
It will also be more flexible in future to use. It might also better tie in together with these other places like taking `eng/common` from a different place etc. That this class could be more vmr-agnostic.